contribute/architecture/state.md

```diff
@@ -0,0 +1,142 @@
+# State
+
+Grafana uses a database to persist settings between restarts. In fact, if you don't specify one, Grafana creates a [SQLite3](https://www.sqlite.org/) database file on your local disk. This guide explains how to store and retrieve data from the database.
+
+Grafana supports the following databases:
+
+- [MySQL](https://www.mysql.com/)
+- [PostgreSQL](https://www.postgresql.org/)
+- [SQLite3](https://www.sqlite.org/)
+
+Grafana uses the [XORM](https://xorm.io) framework for persisting objects to the database. For more information on how to use XORM, refer to the [documentation](http://gobook.io/read/github.com/go-xorm/manual-en-US/).
+
+[Services](services.md) don't use XORM directly. Instead, services use the _SQL store_, a special type of service that provides an abstraction for the database layer. There are two ways of using the `sqlstore`: using `sqlstore` handlers, and using the `SqlStore` instance.
+
+## `sqlstore` handlers
+
+> **Deprecated:** We are deprecating `sqlstore` handlers in favor of using the `SqlStore`  object directly in each service. Since most services still use the `sqlstore` handlers, we still want to explain how they work.
+
+The `sqlstore` package allows you to register command handlers that either store, or retrieve objects from the database. `sqlstore` handlers are similar to services:
+
+- Services are command handlers that _contain business logic_.
+- `sqlstore` handlers are command handlers that _access the database_.
+
+More importantly, command handlers for services are _methods_, while `sqlstore` handlers are _functions_ that access global variables.
+
+### Register a `sqlstore` handler
+
+> **Deprecated:** Refer to the [deprecation note for `sqlstore` handlers](#sqlstore-handlers).
+
+To register a handler:
+
+- Create a new file `myrepo.go` in the `sqlstore` package.
+- Create a [command handler](services.md#handle-commands).
+- Register the handler in the `init` function:
+
+```go
+func init() {
+    bus.AddHandler("sql", SaveDashboard)
```

Comment:
`SaveDashboard` here, but `DeleteDashboard` in the example.

I want to say that we should add a disclaimer about usage of this feature. We don't want users to create and delete over existing tables in their new services as that would make Grafana very hard to maintain. But this doc is perhaps not the right place for that.

Reply:
```suggestion
    bus.AddHandler("sql", DeleteDashboard)
```

Good catch!